app/server.ts

```diff
@@ -1,32 +1,194 @@
+import * as fs from 'node:fs';
+import * as http from 'node:http';
+import * as https from 'node:https';
+import * as os from 'node:os';
 import * as path from 'node:path';
 import { fileURLToPath } from 'node:url';
 import { createRequestHandler } from '@react-router/express';
 import express from 'express';
 
+const dirname = path.dirname(fileURLToPath(import.meta.url));
+const certPath = path.join(dirname, '../certs/localhost-cert.pem');
+const keyPath = path.join(dirname, '../certs/localhost-key.pem');
+
 const viteDevServer =
   process.env.NODE_ENV === 'production'
     ? null
     : await import('vite').then((vite) =>
         vite.createServer({
-          server: { middlewareMode: true },
+          server: {
+            middlewareMode: true,
+            https:
+              fs.existsSync(certPath) && fs.existsSync(keyPath)
+                ? {
+                    key: fs.readFileSync(keyPath),
+                    cert: fs.readFileSync(certPath),
+                  }
+                : undefined,
+          },
         }),
       );
 
 const app = express();
+
+// Proxy Supabase requests in development
+if (process.env.NODE_ENV !== 'production') {
+  app.use('/supabase/*', (req, res) => {
```

Comment:
this proxying is pretty ugly as you have to proxy all http requests and also socket stuff below. it's error prone 

I'll try to find another way